python-package/lightgbm/basic.py

```diff
@@ -21,13 +21,21 @@
 from .libpath import find_lib_path
 
 
+def _log_callback(msg):
+    """Redirect logs from native library into Python console"""
+    print("{0:s}".format(decode_string(msg)), end='')
+
 def _load_lib():
     """Load LightGBM library."""
     lib_path = find_lib_path()
     if len(lib_path) == 0:
         return None
     lib = ctypes.cdll.LoadLibrary(lib_path[0])
     lib.LGBM_GetLastError.restype = ctypes.c_char_p
+    CALLBACK = ctypes.CFUNCTYPE(None, ctypes.c_char_p)
+    lib.callback = CALLBACK(_log_callback)
+    if lib.LGBM_RegisterLogCallback(lib.callback) != 0:
+        raise LightGBMError(decode_string(_LIB.LGBM_GetLastError()))
```

Comment:
Seems that we cannot use `_LIB` here because it leads to circular reference. `lib.LGBM_GetLastError()`?